crates/oxc_linter/src/rules/eslint/sort_imports.rs

```diff
@@ -0,0 +1,835 @@
+                    if current_name < previous_name {
+                        ctx.diagnostic(sort_imports_alphabetically_diagnostic(current.span));
+                    }
+                }
+            }
+            std::cmp::Ordering::Greater => {}
+        }
+    }
+    // Check member sort in a import declaration
+    // ```js
+    // import { b, a } from 'foo.js'
+    // ```
+    fn check_member_sort_with_fix(&self, current: &ImportDeclaration, ctx: &LintContext) {
+        let Some(specifiers) = &current.specifiers else {
+            return;
+        };
+
+        let specifiers: Vec<_> = specifiers
+            .iter()
+            .filter_map(|specifier| {
+                if let ImportDeclarationSpecifier::ImportSpecifier(ref specifier) = specifier {
+                    Some(specifier)
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        if specifiers.len() < 2 {
+            return;
+        }
+
+        let unsorted = specifiers
+            .windows(2)
+            .find(|window| {
+                let a = window[0].local.name.as_str();
+                let b = window[1].local.name.as_str();
+
+                if self.ignore_case {
+                    a.to_lowercase() > b.to_lowercase()
+                } else {
+                    a > b
+                }
+            })
+            .map(|window| (window[1].local.name.as_str(), window[1].span));
+
+        let Some((unsorted_name, unsorted_span)) = unsorted else {
+            return;
+        };
+
+        // ESLint check if there are comments in the ImportSpecifier list, and don't rearrange the specifiers.
+        // We don't have a direct way to check if there are comments in the specifiers(may need lookahead/lookbehind source text).
+        // ```js
+        // // this is not fixable in ESLint
+        // import { /* comment */ a, b, c, d } from 'foo.js'
+        // ```
+        // I use ImportStatement's span to check if there are comments between the specifiers.
+        let is_fixable = !ctx.semantic().trivias().has_comments_between(current.span);
+
+        if is_fixable {
+            // Safe to index because we know that `specifiers` is at least 2 element long
+            let specifiers_span = specifiers[0].span.merge(&specifiers[specifiers.len() - 1].span);
+            ctx.diagnostic_with_fix(
+                sort_members_alphabetically_diagnostic(unsorted_name, unsorted_span),
+                || {
+                    // import { a, b,      c, d } from 'foo.js'
+                    //            ^  ^^^^^^  ^
+                    let mut paddings: Vec<&str> = specifiers
+                        .windows(2)
+                        .map(|window| {
+                            let a = window[0].span;
+                            let b = window[1].span;
+
+                            let padding = Span::new(a.end, b.start);
+                            padding.source_text(ctx.source_text())
```

Comment:
```suggestion
                            ctx.source_range(padding)
```

Reply:
Maybe we should do a big replacement for all the rules in a separate PR as well?